src/firefly/js/ui/tap/TableSearchMethods.jsx

```diff
@@ -161,103 +123,290 @@ function isPanelChecked(searchItem, fields) {
-                {false && <WavelengthSearch {...{cols, groupKey, fields, initArgs:this.props.initArgs}} />}
-            </FieldGroup>
-        );
-    }
+function useSiaReducer(component, siaReducer) {
+    const [siaQueryResult, setSiaQueryResult] = useState(null);
+
+    useEffect(() => {
+        function reduceQuery(fields, newFields) {
+            const siaQueryResult = siaReducer(fields, newFields);
+            setSiaQueryResult(siaQueryResult);
+            return siaQueryResult;
+        }
+        constraintReducers.siaReducerMap.set(component, reduceQuery);
+        return () => {
+            constraintReducers.siaReducerMap.delete(component);
+        };
+    });
+
+    return siaQueryResult;
 }
 
+function useFieldGroupReducer(component, fieldGroupReducer) {
+    useEffect(() => {
+        chainedReducers.set(component, fieldGroupReducer);
+        return () => {
+            chainedReducers.delete(component);
+        };
+    });
+}
 
-const showTimePicker = (loc, show) => {
-    const timeKey = loc === FROM ? TimeFrom : TimeTo;
-    const title = loc === FROM ? 'select "from" time' : 'select "to" time';
-    const valueInfo = getTimeValueInfo(ISO, skey, timeKey);
-    const pickerKey = get(timeKeyMap, [timeKey, ISO]);
+const FunctionalTableSearchMethods = (props) => {
+
+    const [fields, setFields] = useState(FieldGroupUtils.getGroupFields(skey));
+    //const [queryState, dispatchQueryState] = useReducer();
+    // FIXME: Rationalize use of state, props, etc...
+    //const [columnsModel, setColumnsModel] = useState(props ? props.columnsModel : getTblById(get(fields, [CrtColumnsModel, 'value'])));
+    const columnsModel = props.columnsModel;
+    const groupKey = skey;
+    const [adqlResults, setAdqlResults] = useState();
+
+    useEffect(() => {
+        setFields(FieldGroupUtils.getGroupFields(skey));
+        return FieldGroupUtils.bindToStore(groupKey, (fields) => {
+            const {adqlResults} = fields;
+            setAdqlResults(adqlResults);
+        });
+    }, []);
 
-    const content = (
-            <DateTimePickerField fieldKey={pickerKey}
-                                 groupKey={skey}
-                                 showInput={false}
-                                 openPicker={true}
-                                 value={valueInfo.value}
-                                 inputStyle={{marginBottom: 3}}
-            />);
+    useEffect(() => {
+        // We effectively want to invalidate the computed ADQL results when the
+        // constraint reducers change. Dispatching an update will trigger the reducers.
+        dispatchValueChange({...{value: ''}, fieldKey: AdqlResults, groupKey});
+    }, [constraintReducers]);
 
-    showOptionsPopup({content, title, modal: true, show});
+    const obsCoreEnabled = props.obsCoreEnabled;
+    const cols = getAvailableColumns(columnsModel);
+    const DEBUG_OBSCORE = get(getAppOptions(), ['obsCore', 'debug'], false);
```

Comment:
we are trying to get away from lodash `get`
```js
const DEBUG_OBSCORE = getAppOptions().obsCore?.debug ?? false;
```

Reply:
Thanks, I'll reorganize this